### PR TITLE
[receiver/dockerstats] remove deprecated container.cpu.percent and complete the transition to container.cpu.utilization

### PR DESCRIPTION
**Pull request by sigilioso**
**Description:** <Describe what has changed.>

Following up #24183, this PR removes the deprecated `container.cpu.percent` metric as explained in README's [deprecation section](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/dockerstatsreceiver#transition-to-cpu-utilization-metric-name-aligned-with-opentelemetry-specification).

**Link to tracking Issue:** #21807

**Testing:** <Describe what testing was performed and which tests were added.>

```
❯ go test -race -timeout 300s -parallel 4 --tags="" ./... -count 1
ok      github.com/open-telemetry/opentelemetry-collector-contrib/receiver/dockerstatsreceiver  1.480s
ok      github.com/open-telemetry/opentelemetry-collector-contrib/receiver/dockerstatsreceiver/internal/metadata        1.525s
```

**Documentation:** the documentation has been updated using `mdatagen`

### CODE REVIEW

File: .chloggen/remove-deprecated-cpu-metric.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: deprecation
```
Comment (mx-psi) [inline .chloggen/remove-deprecated-cpu-metric.yaml:4]: We already deprecated this metric, this is a breaking change:

```suggestion
change_type: breaking
```

**Review by mx-psi — COMMENTED** [review]
The PR title should talk about `cpu.container.percent` instead, shouldn't it?

**Review by mx-psi — APPROVED** [review]

**Comment by sigilioso** [comment]
> The PR title should talk about `cpu.container.percent` instead, shouldn't it?

I forgot to edit the PR title 🤦, thanks for pointing it out.  I've just changed it.

**Comment by dmitryax** [comment]
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24183 has not been released yet. Please update the schedule in readme in a separate PR. We should hold off on merging this one until 0.88.0 is released

**Review by dmitryax — CHANGES_REQUESTED** [review]
LGTM but we need to hold off until 0.88.0 is released

**Comment by sigilioso** [comment]
The schedule has just been updated in #27853 andI will rebase this when it is merged. Thanks for clarifying!

**Review by dmitryax — APPROVED** [review]

**Pull request merged**